src/main/java/org/opentripplanner/middleware/controllers/api/ApiUserController.java

```diff
@@ -47,29 +47,39 @@ protected void buildEndpoint(ApiEndpoint baseEndpoint) {
         LOG.info("Registering path {}.", ROOT_ROUTE + ID_PATH + API_KEY_PATH);
         ApiEndpoint modifiedEndpoint = baseEndpoint
             // Create API key
-            .post(path(ID_PATH + API_KEY_PATH)
+            .post(MethodDescriptor.path(ID_PATH + API_KEY_PATH)
                     .withDescription("Creates API key for ApiUser (with optional AWS API Gateway usage plan ID).")
                     .withPathParam().withName(ID_PARAM).withRequired(true).withDescription("The user ID")
                     .and()
                     .withQueryParam().withName("usagePlanId").withDescription("Optional AWS API Gateway usage plan ID.")
                     .and()
-                    .withProduces(JSON_ONLY)
+                    .withProduces(HttpUtils.JSON_ONLY)
                     .withResponseType(persistence.clazz),
                 this::createApiKeyForApiUser, JsonUtils::toJson
             )
             // Delete API key
-            .delete(path(ID_PATH + API_KEY_PATH + API_KEY_ID_PARAM)
+            .delete(MethodDescriptor.path(ID_PATH + API_KEY_PATH + API_KEY_ID_PARAM)
                     .withDescription("Deletes API key for ApiUser.")
                     .withPathParam().withName(ID_PARAM).withDescription("The user ID.")
                     .and()
                     .withPathParam().withName("apiKeyId").withDescription("The ID of the API key.")
                     .and()
-                    .withProduces(JSON_ONLY)
+                    .withProduces(HttpUtils.JSON_ONLY)
                     .withResponseType(persistence.clazz),
-                this::deleteApiKeyForApiUser, JsonUtils::toJson
+                this::deleteApiKeyForApiUser, JsonUtils::toJson)
+            // Authenticate user with Auth0
+            .post(MethodDescriptor.path(ID_PATH + AUTHENTICATE_PATH)
+                    .withDescription("Authenticates ApiUser with Auth0.")
+                    .withPathParam().withName(ID_PARAM).withDescription("The user ID.").and()
+                    .withQueryParam().withName(USERNAME_PARAM).withRequired(true)
+                    .withDescription("Auth0 username (usually email address).").and()
+                    .withQueryParam().withName(PASSWORD_PARAM).withRequired(true)
```

Comment:
Hmm... not sure how I feel about passing credentials as query params (as in `/authenticate?username=xxx&password=yyy`)

Reply:
I'm not sure on it either, but can't authenticate without it.